File: 2-2.py
import pandas as pd

df = pd.read_csv('pandas\edu_bigdata_imp1.csv', encoding = 'big5', low_memory = False)
df_filtered = df[df['PseudoID'] == 281]
df_filtered = df_filtered[['dp001_question_sn', 'dp001_prac_score_rate']]
unique_value = df_filtered[df_filtered['dp001_prac_score_rate'] == 100]
print(len(unique_value))

File: 2-2.php
<?php

$filename = 'pandas/edu_bigdata_imp1.csv';

$csvData = array_map('str_getcsv', file($filename));
$header = array_shift($csvData);

$pseudoIDIndex = array_search('PseudoID', $header);
$questionSNIndex = array_search('dp001_question_sn', $header);
$pracScoreRateIndex = array_search('dp001_prac_score_rate', $header);

$filteredValues = [];

foreach ($csvData as $row) {
    if ($row[$pseudoIDIndex] == 281 && $row[$pracScoreRateIndex] == 100) {
        $filteredValues[] = [$row[$questionSNIndex], $row[$pracScoreRateIndex]];
    }
}

$filteredCount = count($filteredValues);

echo $filteredCount;

?>
